Paintroid/src/androidTest/java/org/catrobat/paintroid/test/espresso/tools/StampToolIntegrationTest.java

```diff
@@ -203,6 +204,34 @@ public void testCopyPixel() {
 		assertEquals("Second Pixel not Black after using Stamp for copying", Color.BLACK, pixelToControl);
 	}
 
+	@Test
+	public void testStampToolNotCapturingOtherLayers() {
+		PointF surfaceCenterPoint = getScreenPointFromSurfaceCoordinates(perspective.surfaceCenterX, perspective.surfaceCenterY);
+		onView(isRoot()).perform(touchAt(surfaceCenterPoint.x, surfaceCenterPoint.y - Y_CLICK_OFFSET));
+
+		onToolBarView()
+				.performSelectTool(ToolType.STAMP);
+
+		StampTool stampTool = (StampTool) toolReference.get();
+		PointF toolPosition = new PointF(surfaceCenterPoint.x, surfaceCenterPoint.y - Y_CLICK_OFFSET);
+		stampTool.toolPosition.set(toolPosition);
+
+		onLayerMenuView()
+				.performOpen()
+				.performAddLayer();
+
+		onLayerMenuView()
+				.performClose();
+
+		onDrawingSurfaceView()
+				.perform(touchAt(DrawingSurfaceLocationProvider.TOOL_POSITION, tapStampLong));
```

Comment:
you copy the area but don't paste it, so the check below does checks always an transparent pixel.